graph/generator.js

```diff
@@ -79,55 +54,43 @@ function getParents(typeName, typeList) {
   });
 }
 
-function jsonSanitizer(key, value) {
-  if (key === 'isBuiltin') {
-    /* eslint-disable no-undefined */
-    return undefined;
-    /* eslint-enable no-undefined */
-  }
-
-  return value;
-}
-
 function extractTypeData(types) {
-  return types.map(type => {
-    const allFields = type.fields || [];
-
-    const fields = allFields.filter(field => {
-      return (isBuiltin(getBaseType(field.type)) && field.args.length === 0);
+  return types.filter(type => {
+    return type.kind === 'OBJECT';
+  }).map(type => {
+    const scalars = type.fields.filter(field => {
+      return isScalar(getBaseType(field.type));
     });
 
-    const fieldsWithArgs = allFields.filter(field => {
-      return (isBuiltin(getBaseType(field.type)) && field.args.length < 0);
+    const objects = type.fields.filter(field => {
+      const baseType = getBaseType(field.type);
+
+      return isObject(baseType) && !isConnection(baseType);
     });
 
-    const relationships = allFields.filter(field => {
-      return !isBuiltin(getBaseType(field.type));
+    const connections = type.fields.filter(field => {
+      return isConnection(getBaseType(field.type));
     });
 
     return {
       name: type.name,
-      isBuiltin: isBuiltin(type.name),
-      fields: fields.map(transformField).reduce(objectifyField, {}),
-      fieldsWithArgs: fieldsWithArgs.map(transformFieldWithArgs).reduce(objectifyField, {}),
-      relationships: relationships.map(transformRelationship).reduce(objectifyField, {}),
+      kind: type.kind,
+      scalars: scalars.map(transformField).reduce(objectifyField, {}),
+      objects: objects.map(transformField).reduce(objectifyField, {}),
+      connections: connections.map(transformField).reduce(objectifyField, {}),
```

Comment:
When creating these arrays of types (`objects`, etc), `getBaseType` was already run on them. Then `transformField` ends up doing the same thing again. Is possible to store the the base type on the first pass and then skip the 2nd one?